src/main/java/org/manifold/compiler/front/ExpressionEdge.java

```diff
@@ -30,8 +33,7 @@ public void setTarget(ExpressionVertex newTarget) {
 
   public ExpressionEdge(ExpressionVertex from, ExpressionVertex to) {
     if (from == null && to == null) {
-      throw new UndefinedBehaviourError(
-          "attempt to create a totally disconnected ExpressionEdge");
+      log.warn("attempt to create a totally disconnected ExpressionEdge");
     }
     source = from;
     target = to;
```

Comment:
Yeah, I know in at least a couple of cases the source or target will be changed. One great example is the rest of the stuff happening in this code.